google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -101,27 +100,28 @@ public int read(ByteBuffer dst) throws IOException {
       checkOpen();
       int amt;
       int retries = 0;
-      int maxRetries = 3;
+      int maxRetries = Math.max(3, maxChannelReopens);
```

Comment:
maxChannelReopens is 0 by default, 3 in GATK (the application that's motivating this code in the first place). If a user is willing to reopen the channel more often, then it's probably also OK to retry normal errors more often.